compiler/qsc_frontend/src/compile.rs

```diff
@@ -210,6 +210,12 @@ pub struct PackageStore {
     next_id: PackageId,
 }
 
+impl Debug for PackageStore {
```

Comment:
Can we impl `Display` instead? It feels unconventional to have the `Debug` emit a "lossy" representation - I would expect it to display every bit of data in a struct. But it's entirely possible I'm making this up since I can't really find any documentation to back up my opinion...

Reply:
I'm with you on `Display` but we already break that rule all over the place for HIR/AST types - the `Display` impls we have are more of a developer-facing representation, so [we ended up adding another `display()` method](https://github.com/microsoft/qsharp/blob/09ded186cf91547ac1c0701d287b5c6b7fe8102e/compiler/qsc_hir/src/ty.rs#L68) that _is_ for actual users . It's already a bit messy, I'm okay whichever way you choose to go on this.